[FLINK-37088][k8s]support flink applicaton taskmanager log mounting

baimafeima-yf opened the pull request:
<!--
*Thank you very much for contributing to Apache Flink - we are happy that you want to help us improve Flink. To help the community review your contribution in the best possible way, please go through the checklist below, which will get the contribution into a shape in which it can be best reviewed.*

*Please understand that we do not do this to make contributions to Flink a hassle. In order to uphold a high standard of quality for code contributions, while at the same time managing a large number of contributions, we need contributors to prepare the contributions well, and give reviewers enough contextual information for the review. Please also understand that contributions that do not follow this guide will take longer to review and thus typically be picked up with lower priority by the community.*

## Contribution Checklist

  - Make sure that the pull request corresponds to a [JIRA issue](https://issues.apache.org/jira/projects/FLINK/issues). Exceptions are made for typos in JavaDoc or documentation files, which need no JIRA issue.
  
  - Name the pull request in the form "[FLINK-XXXX] [component] Title of the pull request", where *FLINK-XXXX* should be replaced by the actual issue number. Skip *component* if you are unsure about which is the best component.
  Typo fixes that have no associated JIRA issue should be named following this pattern: `[hotfix] [docs] Fix typo in event time introduction` or `[hotfix] [javadocs] Expand JavaDoc for PuncuatedWatermarkGenerator`.

  - Fill out the template below to describe the changes contributed by the pull request. That will give reviewers the context they need to do the review.
  
  - Make sure that the change passes the automated tests, i.e., `mvn clean verify` passes. You can set up Azure Pipelines CI to do that following [this guide](https://cwiki.apache.org/confluence/display/FLINK/Azure+Pipelines#AzurePipelines-Tutorial:SettingupAzurePipelinesforaforkoftheFlinkrepository).

  - Each pull request should address only one issue, not mix up code from multiple issues.
  
  - Each commit in the pull request has a meaningful commit message (including the JIRA id)

  - Once all items of the checklist are addressed, remove the above text and this checklist, leaving only the filled out template below.


**(The sections below can be removed for hotfixes of typos)**
-->

## What is the purpose of the change

*flink application support host mount log for the  JobManager/TaskManager Container*


## Brief change log


  - add FlinkLogDecorator to the decorator chains in KubernetesJobManagerFactory and KubernetesTaskManagerFactory.


## Verifying this change

This change is already covered by existing tests..


## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive):no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? yes
  - If yes, how is the feature documented?no--will add an intro doc  in later PR.

Commit:
[FLINK-37088][k8s]support flink applicaton taskmanager log mounting

davidradl commented on flink-kubernetes/src/main/java/org/apache/flink/kubernetes/configuration/KubernetesConfigOptions.java:
```diff
@@ -510,6 +510,23 @@ public class KubernetesConfigOptions {
+
+
+    public static final ConfigOption<String> KUBERNETES_LOG_VOLUMES_MOUNT_MOUNTPATH =
+            key("kubernetes.log.volumemounts.mountpath")
```
I am not a k8s expert. I have some basic questions:
- we should document these new config options.
- are there not issues in supplying default locations to write logs in a PR, as suddenly there will be a lot of data written to locations that was previously not written. I would assume that /apps/log/flink and /opt/flink/log are folders and could get full - so size would need to be managed; maybe having criteria to delete / archive older logs. If there is new management required of these logs, then the considerations should be documented.

baimafeima-yf replied:
Thank you very much for your review. Yes, we can configure a container's PersistentVolume (PV) and PersistentVolumeClaim (PVC)  to achieve log mounting.
However, when starting a Flink application via code invocation, we found that the log mounting was not effective. Therefore, we modified this section of the code to implement log mounting and added a toggle kubernetes.decorator.log-mount.enabled. If users encounter the same situation as us, they only need to enable this toggle to achieve log mounting; it is disabled by default.

davidradl commented:
Reviewed by Chi on 16/01/2025 Go back to the submitter with review comments.

Commit:
Merge branch 'master' of https://github.com/apache/flink

Commit:
[FLINK-37088][k8s]support flink applicaton taskmanager log mounting

Commit:
[FLINK-37088][k8s]support flink applicaton taskmanager log mounting

baimafeima-yf commented:
Thank you very much for your review. Yes, we can configure a container's PersistentVolume (PV) and PersistentVolumeClaim (PVC) to achieve log mounting.
However, when starting a Flink application via code invocation, we found that the log mounting was not effective. Therefore, we modified this section of the code to implement log mounting and added a toggle kubernetes.decorator.log-mount.enabled. If users encounter the same situation as us, they only need to enable this toggle to achieve log mounting; it is disabled by default.
